[Solutions] Local&Side Nav 컴포넌트와 제품 및 특징, 기능, 기대효과 컴포넌트 초안

ryudg opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?
> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- Solutions 페이지의 기능 및 레이아웃 (header, main, nav, footer 등 공통 section 을 제외) 을 구현하였습니다.


## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?
> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- 테블릿, 데스크탑, 모바일 반응형 레이아웃 구현
- LNB 컴포넌트로 각각 솔루션들의 링크로 이동
- SNB 컴포넌트로 현재 페이지의 해당 섹션으로 이동
- 제품 및 특징, 기능, 기대효과를 더미 데이터(Json)을 이용해 불러와 리스트 렌더링

https://github.com/ReOpenSG/ReOpenSG/assets/103430498/9d2050f4-2019-4040-806c-54986d2e0a81

https://github.com/ReOpenSG/ReOpenSG/assets/103430498/4b81fa9a-4ac7-4e40-9887-f0bf79677694



## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?
> 없으면 "없음" 이라고 기재해 주세요

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요
> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

- `<p className={styles.nav}>{prevLocation} &nbsp; &gt; &nbsp; {currentLocation}</p>` ➡️ ` > `
  - Solution Banner 상단의 현재 경로를 알려주는 텍스트를 `>`로 표기하는게 맞을지?!
![image](https://github.com/ReOpenSG/ReOpenSG/assets/103430498/c0364bd3-9340-42c7-b0b3-206c9baa523a)

- Lnb(Snb 동일) 컴포넌트의 item을 클릭하면 url이 변경되며 하위 컨텐츠의 정보가 현재 url을 기반으로 어떤 솔루션인지를 파악해 컨텐츠의 정보를 렌더링 합니다.

<img width="744" alt="image" src="https://github.com/ReOpenSG/ReOpenSG/assets/103430498/be3b03f1-4ec8-4e80-b680-82958a6e7b6d">

이때, url 주소가 변경됨에 따라 이전 페이지(뒤로가기)를 반복해서 눌러야 하는 상황이 발생합니다. 이를 해결하기 위한 방법이 있을까요?



## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?
> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요
- 없음

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

ryudg committed:
➕ [Add] : Solutions Json Data 추가
[데이터](https://docs.google.com/spreadsheets/d/1EJYMvt6raDFlVyOrO7gNiJ_jIjcFLPtYhbX0JLln3Jg/edit?usp=sharing "OpenSG Solutions 엑셀 데이터 구글 스프레드 시트")를 기반으로 `Soulitions.json` 데이터 생성

ryudg committed:
➕ [Add] : 솔루션 LNB 링크 데이터 렌더링
저장된 솔루션 json 파일에서 데이터를 가져와 동일한 솔루션을 가진 제품들로 LNB 렌더링

ryudg committed:
➕ [Add] : TailwindCSS 공통 설정 추가
- TailwindCSS 공통 설정 추가
 1. colors
 2. fontFamily
 3. fontSize
 4. fontWeight
 5. screens
 6. maxWidth, minWidth
 7. gap

ryudg committed:
➕ [Add] : Lnb, Banner, Char 컴포넌트 분리
- Solutions 컴포넌트에서 Lnb, Banner, Char 컴포넌트 분리
- Solutions 컴포넌트에서 Data를 받아와서 각 컴포넌트에 props로 전달

ryudg committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

ryudg committed:
📂 [Rename] : Banner 컴포넌트명 변경
Banner 컴포넌트 이름을 SolutionBanner로 변경하였습니다.

ryudg committed:
✨ [Feat] : Lnb 컴포넌트 레이아웃 구현
- 반응형 레이아웃 구현
  - 모바일 화면에서 텍스트 변경
- NavLink를 이용한 페이지 이동 구현
  - 해당하는 페이지로 이동 시, 해당 페이지의 Lnb 메뉴 활성화

Github issue #8

ryudg committed:
🐛 [Fix] : Lnb 링크 색상 변경 Transition 수정
- Lnb 링크 색상 변경 Transition 수정
  - 기존 transion-all에서 transition-colors로 변경

Github issue #8

ryudg committed:
✨ [Feat] : Solutions views 반응형 디자인 추가
- 모바일, 테블릿, 데스크탑 별 여백 및 레이아웃 변경 기능 추가

Github issue #5

ryudg committed:
🐛 [Fix] : 반응형 breakpoint 수정
- 기존 breakpoint에서 수정
  - 기본 스타일을 mobile로 설정 한 후, tablet, desktop일 때 스타일 적용

Github issue #5, #8

ryudg committed:
✨ [Feat] : Snb 컴포넌트, Char(제품 및 특징) 컴포넌트 작성
- 반응형 레이아웃 구현
- 각 섹션별 타이을 작성(sr-only를 이용하여 스크린리더에 읽히지 않도록 함)
- 기타 반응형 디자인 수정

Github issue #9, #11

ryudg committed:
➕ [Add] : Func(기능) 컴포넌트 작성
- 반응형 레이아웃 구성
- placeholder 이미지 추가

Github issue #10

ryudg committed:
➕ [Add] : Effect(기대효과) 컴포넌트 작성
- 반응형 레이아웃 구현

Github issue #12

ryudg committed:
✨ [Feat] : Snb 포지션 sticky로 변경
- 사용자 스크롤 움직임에 따라 Snb가 따라오도록 변경

Github issue #11

ryudg committed:
⚙️ [Chore] : react-intersection-observer 패키지 설치
Scrollspy 이벤트를 위한 `react-intersection-observer` 패치키 추가

Github issue #11

ryudg committed:
🐛 [Fix] : `react-intersection-observer` Scrollspy 오류 해결
- 섹션별 스크롤 이벤트를 위해 `react-intersection-observer` 라이브러리를 사용했으나,
  섹션별로 중첩되는 문제가 발생하여 조건문을 활용해 해결.
- gap으로 여백 설정 시 Snb 링크를 클릭하면 해당 섹션으로 이동하지만,
  Header가 가리는 문제가 발생하여 `pt` 속성을 활용해 해결.

Github issue #11

ryudg committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

ryudg committed:
🌈 [Design] : Components css TailwindCSS module로 분리
- Components의 TailwindCSS를 module로 분리

Github issue #5

ryudg committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

ryudg committed:
🌈 [Design] : Tailwind 변수 추가로 인한 css 변경
- tailwind.config.js에 spacing 변수 추가로 인한 css 변경
  - 기존 `[]px` -> 설정해 둔 open 변수로 변경
- 절대 경로 설정

Github issue #5

ryudg committed:
✨ [Feat] : uuidv4를 이용한 고유한 key값 부여
- 기존 key값은 index를 이용하여 부여하였으나, 이는 key값이 중복될 수 있음
  - uuidv4를 이용하여 고유한 key값을 부여함

Github issue #5

dani-day commented on src/components/Solutions/LNB.jsx:
P1: LNB.jsx 만 다른 컴포넌트와 네이밍컨벤션이 다르게 적용된 것 같습니다. 컴포넌트 안에서도 Lnb 로 작성해주신 것 처럼 파일명도 동일하게 변경되면 좋을 것 같습니다 😄

ryudg replied:
Rename으로 직접 파일명 변경 시 git이 파일명을 인식하지 못하는 문제가 있어서 `git mv LNB.jsx Lnb.jsx` 명령어로 변경 완료했습니다.

dani-day commented on src/views/Solutions.jsx:
P2: Soultions.jsx 내에 작성되어있는 css 에서는 tailwindcss 에 spacing 설정을 해두었던 부분이 반영되지 않은 부분이 있어보입니다.

```html
<section className="b-[100px] w-full h-full desktop:px-[64px] tablet:px-[64px] px-[16px]">
```

open-gutter-mobile, open-gutter-desktop 등으로 변경되어야 추후 spacing 조절 시에 전체 적용되지 않을까 싶은데 이유가 있어 남겨두신 건가 싶어서 P2 로 의견 남기겠습니다.

ryudg replied:
`src/views/Solutions.jsx` Solutions 컴포넌트의 tailwindcss를 tailwind.config.js에 적용되어있는 css로 변경하였습니다.

https://github.com/ReOpenSG/ReOpenSG/blob/887e1602565155157bb0f204782b3b6333d5ced1/src/views/Solutions.jsx#L33-L47

dani-day reviewed (changes requested):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다 😸
- LNB.jsx (P1 review)
- Solutions.jsx (P2 review)

ryudg committed:
🐛 [Fix] : LNB.jsx 파일명 변경
- LNB.jsx 파일명 변경
  - LNB.jsx -> Lnb.jsx
    - Rename으로 파일명 변경시 git이 파일명을 인식하지 못하는 문제가 있어서 `git mv LNB.jsx Lnb.jsx` 명령어로 변경

ryudg committed:
🐛 [Fix] : Solutions.jsx TailwindCSS 변경
- 기존 `-[]`로 작성되어 있던 스타일을 tailwind.config.js에 작성된 변수 스타일로 변경

Github issue #5

kelvin-chihyun commented on src/routes/routes.jsx:
P3: 주석처리하신 부분 삭제바랍니다!

kelvin-chihyun reviewed (approved):
확인했고, 큰 이상 없어 보입니다.

ingbinsee commented on src/components/Solutions/Lnb.jsx:
P1: 뒤로가기 버튼 클릭 시 이전 solutions를 거쳐야만 원하는 페이지로 이동 가능합니다. 해당 부분 수정 요청합니다.

ryudg replied:
해당 부분 이후 리팩토링 진행하면서 아이디어 나오는데로 수정하겠습니다.

ingbinsee reviewed (approved):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다.

Lnb.jsx (P1 review)

ingbinsee reviewed (changes requested):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다.

Lnb.jsx (P1 review)

ryudg committed:
💬 [Comment] : 불필요 주석 제거
- `routes/routes.jsx` 파일에 불필요한 주석 제거

Github issue #5

ryudg committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

ryudg committed:
✨ [Feat] : react-router-dom ScrollRestoration
- 산업군에서 제품 링크를 클릭하면 스크롤 위치가 상단이 아닌 이전 위치에 있는 문제를 해결하기 위해
  react-router-dom의 `ScrollRestoration`을 사용해 해결함.

Github issue #5

dani-day reviewed (approved):
수정사항 확인했으며, 추가 리뷰드릴 내용은 없는 것 같습니다 👍
